modules/cas_cache/volume/vol_block_dev_top.c

```diff
@@ -588,7 +588,8 @@ static void _blockdev_set_discard_properties(ocf_cache_t cache,
 		exp_q->limits.discard_granularity =
 			core_q->limits.discard_granularity;
 	} else {
-		blk_queue_max_discard_sectors(exp_q, core_sectors);
+		uint32_t max_discard_size = core_sectors <= UINT_MAX ?: UINT_MAX;
+		blk_queue_max_discard_sectors(exp_q, max_discard_size);
```

Comment:
Will it be any problem if we always set it to `UINT_MAX` regardless of the device size?

Reply:
`UINT_MAX` is set only if the device has more than `UINT_MAX` sectors.